apps/client/src/app/pages/inventory-optimizer/inventory-optimizer/inventory-optimizer.component.ts

```diff
@@ -120,14 +120,41 @@ export class InventoryOptimizerComponent {
   }
 
   constructor(private inventoryFacade: InventoryFacade,
-              @Inject(INVENTORY_OPTIMIZER) private optimizers: InventoryOptimizer[],
-              private lazyData: LazyDataService, private message: NzMessageService, private translate: TranslateService) {
+    @Inject(INVENTORY_OPTIMIZER) private optimizers: InventoryOptimizer[],
+    private lazyData: LazyDataService, private message: NzMessageService, private translate: TranslateService) {
   }
 
   private getContainerName(item: InventoryItem): string {
     return item.retainerName || this.inventoryFacade.getContainerName(item.containerId);
   }
 
+  public getExpansions(): string[] {
+    const expansions: any[] = this.lazyData.patches.map(p => {
+      return { name: p.ExName, version: p.ExVersion }
+    });
+
+    return expansions.filter((value, index, array) => {
+      return index === array.findIndex(e => e.name == value.name);
```

Comment:
```suggestion
      return index === array.findIndex(e => e.name === value.name);
```

Reply:
I replaced this with `uniqBy` so the code mentioned here no longer exists.